src/core/manager.c

```diff
@@ -2416,6 +2442,12 @@ static int manager_dispatch_notify_fd(sd_event_source *source, int fd, uint32_t
         /* Make sure it's NUL-terminated. */
         buf[n] = 0;
 
+        if (fds) {
+                /* possibly a barrier fd, let's see */
+                if (manager_process_barrier_fd(buf, fds))
```

Comment:
hmm, why the separate `fds` check beforehand? sounds unnecessary and is done in the function itself again

Reply:
fds is NULL if 0 fds are passed (afaics), the check can be moved to the call, but then we end up calling it everytime we get a notification...

Reply:
which is fine. given the function is called only from one place anyway, the compiler should inline it anyway, and thus there should be zero difference effectively